samples/drivers/soc_flash_nrf/src/main.c

```diff
@@ -67,16 +64,12 @@ int main(void)
 {
 	const struct device *flash_dev = TEST_PARTITION_DEVICE;
 	struct flash_parameters flash_params;
-	uint32_t buf_array_1[4] = { TEST_DATA_WORD_0, TEST_DATA_WORD_1,
-				    TEST_DATA_WORD_2, TEST_DATA_WORD_3 };
-	uint32_t buf_array_2[4] = { TEST_DATA_WORD_3, TEST_DATA_WORD_1,
-				    TEST_DATA_WORD_2, TEST_DATA_WORD_0 };
-	uint32_t buf_array_3[8] = { TEST_DATA_WORD_0, TEST_DATA_WORD_1,
-				    TEST_DATA_WORD_2, TEST_DATA_WORD_3,
-				    TEST_DATA_WORD_0, TEST_DATA_WORD_1,
-				    TEST_DATA_WORD_2, TEST_DATA_WORD_3 };
-	uint32_t buf_word = 0U;
+
+	const uint8_t test_data[16] = {0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08,
+				       0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F, 0x10};
+	uint8_t read_buffer[16] = {0};
```

Comment:
Do an identifier for the 16, so that we can easily track and amend the size here if needed.